source/common/common/regex.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <memory>
+#include <regex>
+
+#include "envoy/common/regex.h"
+#include "envoy/type/matcher/regex.pb.h"
+
+namespace Envoy {
+namespace Regex {
+
+/**
+ * Utilities for constructing regular expressions.
+ */
+class Utility {
+public:
+  /**
+   * Constructs a std::regex, converting any std::regex_error exception into an EnvoyException.
+   * @param regex std::string containing the regular expression to parse.
+   * @param flags std::regex::flag_type containing parser flags. Defaults to std::regex::optimize.
+   * @return std::regex constructed from regex and flags.
+   * @throw EnvoyException if the regex string is invalid.
+   */
+  static std::regex parseStdRegex(const std::string& regex,
```

Comment:
Can we structurally ban std::regex outside a whitelist with check_format as well?

Reply:
Yes I will look at that.